hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -666,6 +666,25 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 	}
 	meta.SetStatusCondition(&hcluster.Status.Conditions, hyperutil.GenerateReconciliationActiveCondition(hcluster.Spec.PausedUntil, hcluster.Generation))
 
+	// Set ValidOIDCConfiguration condition
+	{
+		condition := meta.FindStatusCondition(hcluster.Status.Conditions, string(hyperv1.ValidOIDCConfiguration))
```

Comment:
can you please move this logic to line 1331 within the aws block where set this and make user we don't gate on if meta.IsStatusConditionFalse(hcluster.Status.Conditions, string(hyperv1.ValidOIDCConfiguration)) {

Reply:
@jparrill This issue is about fixing the logic in l1331 which right now only sets ValidOIDCConfiguration=true if it was previously set to false for any reason. See l1343 check `if meta.IsStatusConditionFalse(hcluster.Status.Conditions, string(hyperv1.ValidOIDCConfiguration)) {` needs to be removed so the condition is always set for the happy path.
We don't need to set it up here, if something fails before ValidOIDCConfiguration was even evaluated that's fine. We can later on evaluate to set it earlier if we think there's any value, for now let's keep it simple and just fix the unnecessary check.